projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts

```diff
@@ -22,4 +23,99 @@ test.describe('ComboBox', () => {
+
+                await tuiGoto(
+                    page,
+                    `components/combo-box/API?strict=${strict}&sandboxExpanded=true`,
+                );
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-${strict}.png`,
+                );
+
+                await textfield.click();
+                await page.keyboard.type('Rubles (500)');
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-waited-mark-${strict}.png`,
+                );
+
+                await comboBoxPO.selectOptions([0]);
+                await apiPageExample.click({force: true});
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-focused-${strict}.png`,
+                );
+
+                await textfield.click();
+                await page.keyboard.press('Backspace');
+                await apiPageExample.click({force: true});
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-backspaced-${strict}.png`,
+                );
+
+                await textfield.click();
+                await page.keyboard.press('Control+A');
+                await page.keyboard.press('Backspace');
+                await apiPageExample.click({force: true});
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-remove-all-${strict}.png`,
+                );
+            });
+
+            test(`correct word match when strict is ${strict}`, async ({page}) => {
+                const {apiPageExample} = new TuiDocumentationPagePO(page);
+                const comboBoxPO = new TuiComboBoxPO(apiPageExample);
+                const textfield = comboBoxPO.textfield.first();
+
+                await tuiGoto(
+                    page,
+                    `components/combo-box/API?strict=${strict}&sandboxExpanded=true`,
+                );
+
+                await textfield.click();
+                await page.keyboard.type('dOlLaRs (237)');
+                await expect(page).toHaveScreenshot(
+                    `correct-word-match-when-strict-${strict}.png`,
+                );
+
+                await textfield.click();
+
+                for (let i = 0; i < 4; i++) {
+                    await page.keyboard.press('Backspace');
+                }
+
+                await expect(page).toHaveScreenshot(
+                    `correct-word-match-when-strict-backspaced-${strict}.png`,
+                );
+
+                await apiPageExample.click({force: true});
+
+                await expect(page).toHaveScreenshot(
+                    `correct-word-match-when-strict-focused-${strict}.png`,
+                );
+
+                await textfield.click();
+                await page.keyboard.press('Control+A');
+                await page.keyboard.press('Backspace');
```

Comment:
In playwright `press('Control+A')` does not work for mac os.
Could we use the following line instead ?

```suggestion
                await textfield.clear();
```

https://playwright.dev/docs/api/class-locator#locator-clear